worldedit-core/src/main/java/com/sk89q/worldedit/util/formatting/text/serializer/gson/StyleSerializer.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * This file is part of text, licensed under the MIT License.
+ *
+ * Copyright (c) 2017-2020 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sk89q.worldedit.util.formatting.text.serializer.gson;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import com.sk89q.worldedit.util.formatting.text.Component;
+import com.sk89q.worldedit.util.formatting.text.event.ClickEvent;
+import com.sk89q.worldedit.util.formatting.text.event.HoverEvent;
+import com.sk89q.worldedit.util.formatting.text.format.Style;
+import com.sk89q.worldedit.util.formatting.text.format.TextColor;
+import com.sk89q.worldedit.util.formatting.text.format.TextDecoration;
+
+import java.lang.reflect.Type;
+
+// This class was copied and manually "relocated" from text3 to update the click/hover events for 1.21.5.
+public class StyleSerializer implements JsonDeserializer<Style>, JsonSerializer<Style> {
+    public static final StyleSerializer INSTANCE = new StyleSerializer();
+
+    private static final TextDecoration[] DECORATIONS = TextDecoration.values();
+
+    static final String COLOR = "color";
+    static final String INSERTION = "insertion";
+    static final String CLICK_EVENT = "click_event";
```

Comment:
How many differences are there between this & the 1.21.4 one? If it's just these constants or a little beyond that- we should be able to fairly trivially just add some version guards here and support <1.21.5

Reply:
there's very few changes in this class. the main issue is that this class has no way to know what MC version is running. usually that's configured by the platform classes, but overriding that would require including a modified version of the platform classes. while we're doing hacky i suppose we could have a public static field in here and have our worldedit platform impls set it to modify the behavior but /shrug.